modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaStorageTypeAwareCheck.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaStorageTypeAwareCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		String newContent = content;
+
+		Matcher deprecatedInterfaceMatcher =
+			_deprecatedInterfacePattern.matcher(content);
+
+		if (deprecatedInterfaceMatcher.find()) {
+			String deprecatedInterface = deprecatedInterfaceMatcher.group();
+
+			if (deprecatedInterface.contains(", StorageTypeAware")) {
+				newContent = StringUtil.removeSubstring(
+					newContent, ", StorageTypeAware");
+			}
+			else if (deprecatedInterface.contains("StorageTypeAware,")) {
+				newContent = StringUtil.removeSubstring(
+					newContent, "StorageTypeAware,");
+			}
+			else {
+				newContent = StringUtil.removeSubstring(
+					newContent, deprecatedInterface);
+			}
+
+			Matcher importStorageTypeAwareMatcher =
+				_importStorageTypeAwarePattern.matcher(newContent);
+
+			if (importStorageTypeAwareMatcher.find()) {
+				newContent = StringUtil.removeSubstring(
+					newContent, importStorageTypeAwareMatcher.group() + "\n\n");
+			}
```

Comment:
@TamyrisTorres, the import of `StorageTypeAware` will not change that much that you need a pattern for it, I'm right? Another thing, do we really need it two `\n`?

What do you think about it we delete this pattern and only do:

```
newContent = StringUtil.removeSubstring(
				"\nimport com.liferay.portal.kernel.scheduler.StorageTypeAware;");

```

Reply:
Please if the remove will return the `newContent` untouched if doesn't find it. But I believe we don't need to check if has it right?  Because if the matcher was triggered means that has the implementation and will have the import, right?

Reply:
Yes, you are correct. 

Resolved!